template.yaml

```diff
@@ -274,3 +274,71 @@ Resources:
         Status: Enabled
+
+  # Athena Data Source
+  AuditAnalysisDatabase:
+    Type: AWS::Glue::Database
+    Properties:
+      CatalogId: !Ref AWS::AccountId
+      DatabaseInput:
+        Description: "TxMA - Event Message Analysis Database"
+        Name: "txma-analysis"
+
+  AuditAnalysisTable:
+    Type: AWS::Glue::Table
+    Properties:
+      CatalogId: !Ref AWS::AccountId
+      DatabaseName: !Ref AuditAnalysisDatabase
+      TableInput:
+        Description: "Table contains event message data to be analysed"
+        Name: !Sub ${AWS::StackName}-txma-audit-sandbox-analysis-table #TODO: Update to !Sub ${AWS::StackName}-txma-audit-${Environment}-analysis-table
+        Parameters: 
+          has_encrypted_data: false
+          projection.enabled: true
+          projection.datetime.type: date
+          projection.datetime.range: '2022/01/01/00,NOW'
+          projection.datetime.format: 'yyyy/MM/dd/HH'
+          projection.datetime.interval: 1
+          projection.datetime.interval.unit: HOURS
+          storage.location.template: !Join ["", ["s3://", !Sub "${AWS::StackName}", "-sandbox-analysis-bucket/firehose/${datetime}/" ]] #TODO: Update to !Join ["", ["s3://", !Sub "${AWS::StackName}", "-", !Sub "${Environment}, "-analysis-bucket/firehose/${datetime}/" ]]
+        PartitionKeys:
+          - {Name: "datetime", Type: string}
+        StorageDescriptor:
+          Columns:
+            - {Name: "event_id", Type: string}
+            - {Name: "client_id", Type: string}
+            - {Name: "timestamp", Type: bigint}
+            - {Name: "timestamp_formatted", Type: string}
+            - {Name: "event_name", Type: string}
+            - {Name: "component_id", Type: string}
+            - {Name: "user", Type: string}
+            - {Name: "platform", Type: string}
+            - {Name: "restricted", Type: string}
+            - {Name: "extensions", Type: string}
+          Compressed: true
+          InputFormat: "org.apache.hadoop.mapred.TextInputFormat"
+          Location: !Sub s3://${AWS::StackName}-sandbox-analysis-bucket/firehose/ #TODO: Update to !Sub s3://${AWS::StackName}-${Environment}-sandbox-analysis-bucket/firehose/
+          OutputFormat: "org.apache.hadoop.hive.ql.io.IgnoreKeyTextOutputFormat"
+          SerdeInfo:
+            Parameters: {"ignore.malformed.json": true,"serialiazation.format": 1,"field.delim":""}
+            SerializationLibrary: "org.openx.data.jsonserde.JsonSerDe"
+          StoredAsSubDirectories: false
+        TableType: EXTERNAL_TABLE
+
+  AthenaAnalysisWorkgroup:
+    Type: AWS::Athena::WorkGroup
+    Properties:
+      Description: The workgroup for TxMA queries
+      Name: TxMAAnalysisQueries
+      WorkGroupConfiguration:
+        EnforceWorkGroupConfiguration: true
+        PublishCloudWatchMetricsEnabled: false
+        ResultConfiguration:
+          EncryptionConfiguration:
+            EncryptionOption: "SSE_S3"
+          OutputLocation:
+            !Join
```

Comment:
Else where we've used the shorthand way of writing these expressions, which personally I find a bit easier to read!

This could be `!Join [ '' [' s3://', !Ref QueryResultsBucket, '// ] ]`

Reply:
Changed as per suggestion